### PR TITLE
Updated search endpoint types

### PR DESCRIPTION
**Pull request by newmoneybigbucks**
I updated some missing fields in the search endpoint   request and result types. I added a interface for paginated results and used the generic result array like you suggested. I added two types for the search query parameters and had other types extend those to reduce some code duplication with common fields.

### CODE REVIEW

File: packages/core/src/types/v3/search/SearchBaseQueryParamters.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export interface BaseQueryParams {
```
Comment (JorrinKievit) [inline packages/core/src/types/v3/search/SearchBaseQueryParamters.ts:1]: Same with this file

**Review by JorrinKievit — COMMENTED** [review]
Thx for your effort!

File: packages/core/src/types/v3/search/search-companies.ts [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
   name: string;
+  origin_country: string;
 }
 
-export interface SearchCompaniesParams {
-  query: string;
-  page?: number;
-}
+// ESLint might complain about this empty interface, but it was left empty (and not removed) so locations that used this type wouldn't break or have to be changed.
+export interface SearchCompaniesParams extends BaseQueryParams {}
```
[the remark this answers is not in the document]
Comment (newmoneybigbucks) [reply packages/core/src/types/v3/search/search-companies.ts:15]: Yeah, ill change all of this right now!

**Comment by JorrinKievit** [comment]
Looks good, will merge it tomorrow!

**Pull request merged**